ggshield/cmd/sca/scan/ci.py

```diff
@@ -65,34 +63,41 @@ def scan_ci_cmd(
     )
 
     config = ContextObj.get(ctx).config
-    try:
-        if not (
-            os.getenv("CI") or os.getenv("JENKINS_HOME") or os.getenv("BUILD_BUILDID")
-        ):
-            raise UsageError("`sca scan ci` should only be used in a CI environment.")
-
-        output_handler = create_output_handler(ctx)
-        if scan_all:
-            result = sca_scan_all(ctx, directory, scan_mode=ScanMode.CI_ALL)
-            scan = SCAScanAllVulnerabilityCollection(id=str(directory), result=result)
-            return output_handler.process_scan_all_result(scan)
+    ci_mode = SupportedCI.from_ci_env()
+    output_handler = create_output_handler(ctx)
 
-        current_commit, previous_commit = get_current_and_previous_state_from_ci_env(
-            config.user_config.verbose
+    try:
+        params = get_scan_ci_parameters(
+            ci_mode, wd=directory, verbose=config.user_config.verbose
         )
+        if params is None:
+            click.echo(
+                "No commit found in merge request, skipping scan.",
+                err=True,
+            )
+            return 0
 
-        ci_mode = SupportedCI.from_ci_env()
+        current_commit, reference_commit = params
         scan_mode = f"{ScanMode.CI_DIFF.value}/{ci_mode.value}"
         result = sca_scan_diff(
             ctx=ctx,
             directory=directory,
-            previous_ref=previous_commit,
+            previous_ref=reference_commit,
             current_ref=current_commit,
             scan_mode=scan_mode,
             ci_mode=ci_mode.name,
         )
         scan = SCAScanDiffVulnerabilityCollection(id=str(directory), result=result)
         return output_handler.process_scan_diff_result(scan)
-
-    except Exception as error:
```

Comment:
Note: the function is already decorated with `@exception_wrapper`, so we can remove the catch-all exception handler